pkg/k8s/apis/cilium.io/client/crds/v2alpha1/ciliumworldcidrsets.yaml

```diff
@@ -0,0 +1,86 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.11.4
+  creationTimestamp: null
+  name: ciliumworldcidrsets.cilium.io
+spec:
+  group: cilium.io
+  names:
+    categories:
+    - cilium
+    kind: CiliumWorldCIDRSet
+    listKind: CiliumWorldCIDRSetList
+    plural: ciliumworldcidrsets
+    singular: ciliumworldcidrset
+  scope: Cluster
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .spec.encapsulate
+      name: Encapsulate
+      type: boolean
+    - jsonPath: .spec.cidrs
+      name: CIDRs
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v2alpha1
+    schema:
+      openAPIV3Schema:
+        description: CiliumWorldCIDRSet is a set of CIDRs and their routing destination.
+          It is used in certain modes (e.g. highscale-IPCache) to indicate whether
+          destinations require encapsulation.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: CiliumWorldCIDRSetSpec configures the routing decision for
+              a set of CIDRs.
+            properties:
+              cidrGroupRefs:
+                description: 'CIDRGroupRefs is a list of CiliumCIDRGroup references.
+                  The encapsulation configuration will apply to all CIDRs referenced
+                  by these groups. Note: Currently only IPv4 is supported.'
+                items:
+                  description: CIDRGroupRef is a reference to a CIDR Group. A CIDR
+                    Group is a list of CIDRs whose IP addresses should be considered
+                    as a same entity when applying fromCIDRGroupRefs policies on incoming
+                    network traffic.
+                  maxLength: 253
+                  pattern: ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
```

Comment:
thought: to the discussion about validation, we could in theory  make this more strict to reject 999 type scenarios at the apiserver level.